guide/api-design-guidelines.md

```diff
@@ -0,0 +1,197 @@
+# Radius API Guidelines
+
+<!-- markdownlint-disable MD033 MD049 MD055 -->
+<!--
+Note to contributors: All guidelines have an anchor tag to allow cross-referencing from associated tooling.
+The anchor tags within a section using a common prefix to ensure uniqueness with anchor tags in other sections.
+Please ensure that you add an anchor tag to any new guidelines that you add and maintain the naming convention.
+-->
+
+## History
+
+<details>
+  <summary>Expand change history</summary>
+
+| Date        | Notes                                                          |
+| ----------- | -------------------------------------------------------------- |
+| 2024-Sep-17 | Added guidance on Secrets                                      |
+
+</details>
+
+## Introduction
+
+These are prescriptive guidelines that Radius contributors MUST follow while designing APIs to maintain a great user experience. These guidelines help make Radius APIs:
```

Comment:
@lakshmimsft, To set the right expectations for readers of this document, I'd recommend being explicit here that currently the guidance is focused on secrets handling. cc @rynowak for inputs as well.

Reply:
updated